src/wallet.rs

```diff
@@ -156,6 +156,44 @@ where
         Ok("0".to_string())
     }
 
+    pub async fn import_ufvk(
+        &mut self,
+        ufvk: &UnifiedFullViewingKey,
+        birthday_height: Option<u32>,
+    ) -> Result<String, Error> {
+        let birthday = match birthday_height {
```

Comment:
would be nice to avoid duplication and factor this logic out between the import methods